File: app/src/main/java/cn/surine/schedulex/data/entity/TimeTableDisplayEntity.java
package cn.surine.schedulex.data.entity;

public class TimeTableDisplayEntity {
    public int session;
    public String startTime;
    public String endTime;

    public TimeTableDisplayEntity(int session, String startTime, String endTime) {
        this.session = session;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    public TimeTableDisplayEntity() {
    }

    public String displayTime() {
        return startTime + "-" + endTime;
    }

    public String displaySession() {
        return "第" + session + "节";
    }


    public TimeTableDisplayEntity setSession(int session) {
        this.session = session;
        return this;
    }

    public TimeTableDisplayEntity setStartTime(String startTime) {
        this.startTime = startTime;
        return this;
    }

    public TimeTableDisplayEntity setEndTime(String endTime) {
        this.endTime = endTime;
        return this;
    }
}

File: app/src/main/java/cn/surine/schedulex/data/entity/TimeTableDisplayEntity.kt
package cn.surine.schedulex.data.entity

class TimeTableDisplayEntity {
    @JvmField
    var session = 0
    @JvmField
    var startTime: String? = null
    @JvmField
    var endTime: String? = null

    constructor(session: Int, startTime: String?, endTime: String?) {
        this.session = session
        this.startTime = startTime
        this.endTime = endTime
    }

    constructor() {}

    fun displayTime(): String {
        return "$startTime-$endTime"
    }

    fun displaySession(): String {
        return "第" + session + "节"
    }

    fun setSession(session: Int): TimeTableDisplayEntity {
        this.session = session
        return this
    }

    fun setStartTime(startTime: String?): TimeTableDisplayEntity {
        this.startTime = startTime
        return this
    }

    fun setEndTime(endTime: String?): TimeTableDisplayEntity {
        this.endTime = endTime
        return this
    }
}